hyperspy/io_plugins/semper_unf.py

```diff
@@ -0,0 +1,520 @@
+                for element in header:
+                    f.write(struct.pack('h', element))  # 2 byte format!
+                f.write(struct.pack('I', 2*len(header)))  # record length!
+                # Write title:
+                if len(self.title) > 0:
+                    f.write(struct.pack('I', len(self.title)))  # record length, 4 byte format!
+                    f.write(self.title)
+                    f.write(struct.pack('I', len(self.title)))  # record length, 4 byte format!
+                # Create label:
+                if self.ilabel:
+                    label = np.zeros(256, dtype=np.int32)
+                    label[:6] = [ord(c) for c in 'Semper']
+                    label[6:8] = divmod(ncol, 256)
+                    label[8:10] = divmod(nrow, 256)
+                    label[10:12] = divmod(nlay, 256)
+                    label[12:14] = divmod(self.iccoln, 256)
+                    label[14:16] = divmod(self.icrown, 256)
+                    label[16:18] = divmod(self.iclayn, 256)
+                    label[18] = self.iclass
+                    label[19] = self.iform
+                    label[20] = self.iwp
+                    year, time = self.date.split(' ')
+                    label[21:24] = map(int, year.split('-'))
+                    label[21] -= 1900
+                    label[24:27] = map(int, time.split(':'))
+                    range_string = '{:.4g},{:.4g}'.format(self.data.min(), self.data.max())
+                    ncrang = len(range_string)
+                    label[27] = ncrang
+                    label[28:28+ncrang] = [ord(c) for c in range_string]
+                    label[55] = self.ipltyp
+                    label[62] = 1  # Use real coords!
+                    label[75:79] = [ord(c) for c in struct.pack('<f', self.scales[2])]  # DZ
+                    label[79:83] = [ord(c) for c in struct.pack('<f', self.offsets[2])]  # Z0
+                    label[83:87] = [ord(c) for c in struct.pack('<f', self.scales[1])]  # DY
+                    label[87:91] = [ord(c) for c in struct.pack('<f', self.offsets[1])]  # Y0
+                    label[91:95] = [ord(c) for c in struct.pack('<f', self.scales[0])]  # DX
+                    label[95:99] = [ord(c) for c in struct.pack('<f', self.offsets[0])]  # X0
+                    label[100:100+len(self.title)] = [ord(s) for s in self.title]
+                    label[244:248] = [ord(c) for c in self.units[0]] + [0]*(4-len(self.units[0]))
+                    label[248:252] = [ord(c) for c in self.units[1]] + [0]*(4-len(self.units[1]))
+                    label[252:256] = [ord(c) for c in self.units[2]] + [0]*(4-len(self.units[2]))
+                    # Write label:
+                    f.write(struct.pack('I', 2*256))  # record length, 4 byte format!
+                    for element in label:
+                        f.write(struct.pack('h', element))  # 2 byte format!
+                    f.write(struct.pack('I', 2*256))  # record length!

```

Comment:
Again, most of this `if self.ilabel: #...` can be moved to a private function that actually creates it (i.e. lines 310-340)